### PR TITLE
tmpl: fix extra newline added on @if-@else-@end block

### LINKED ISSUE #22288: An extra line break when using `@if` in `$tmpl()`
### Describe the bug

`@if` in `$tmpl()` causes an unwanted line break to be inserted.

### Reproduction Steps

Use `@if` in `$tmpl()`, like:

`main.v`
```v
fn main() {
	cond := true
	template := $tmpl('template.txt')
	println(template)
}
```

`template.txt`
```
aaa
@if cond
bbb
@end
ccc
```

### Expected Behavior

I expected

```
aaa
bbb
ccc
```

to be shown

### Current Behavior

It shows

```
aaa

bbb
ccc
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.7 b79aa02

### Environment details (OS name and version, etc.)

V full version: V 0.4.7 c0f7244.b79aa02
OS: macos, macOS, 15.0, 24A335
Processor: 8 cpus, 64bit, little endian, Apple M3

getwd: /Users/syobon/Devel/conditional
vexe: /Users/syobon/.v/v
vexe mtime: 2024-09-23 08:48:38

vroot: OK, value: /Users/syobon/.v
VMODULES: OK, value: /Users/syobon/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.39.5 (Apple Git-154)
Git vroot status: weekly.2024.37-51-gb79aa020
.git/config present: true

CC version: Apple clang version 16.0.0 (clang-1600.0.26.3)
thirdparty/tcc status: thirdparty-macos-arm64 713692d4
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

### PR DESCRIPTION
**Pull request by felipensp**
Fix #22288

### CODE REVIEW

File: vlib/v/tests/tmpl/if_cond.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+@else
+zzz
+@end
+ccc
+
+aaa
+@if cond
+bbb
+@end
+ccc
```
Comment (spytheman) [inline vlib/v/tests/tmpl/if_cond.txt:13]: I think that it would be good, to add a test, that exercises the @else branch as well, because there were changes that affected it in the parser code.

Also please, make it use separate `tmpl_if_cond_test.v` test file. The `tmpl_test.v` file is getting very long, and hard to follow.

Comment (felipensp) [reply vlib/v/tests/tmpl/if_cond.txt:13]: Alright. 👍🏻

**Review by spytheman — APPROVED** [review]
Excellent work 🥳 .
